plugins/modules/identity/keycloak/keycloak_user_federation.py

```diff
@@ -921,12 +921,23 @@ def main():
         after_comp = kc.create_component(desired_comp, realm)
 
         for mapper in updated_mappers:
-            if mapper.get('id') is not None:
-                kc.update_component(mapper, realm)
+            found = kc.get_components(urlencode(dict(parent=cid, name=mapper['name'])), realm)
+            if len(found) > 1:
+                module.fail_json(msg='Found multiple mappers with name `{name}`. Cannot continue.'.format(name=mapper['name']))
+            if len(found) == 1:
+                old_mapper = found[0]
             else:
-                if mapper.get('parentId') is None:
-                    mapper['parentId'] = after_comp['id']
-                mapper = kc.create_component(mapper, realm)
+                old_mapper = {}
+
+            new_mapper = old_mapper.copy()
+            new_mapper.update(mapper)
+
+            if new_mapper.get('id') is not None:
+                kc.update_component(new_mapper, realm)
+            else:
+                if new_mapper.get('parentId') is None:
+                    new_mapper['parentId'] = after_comp['id']
+                mapper = kc.create_component(new_mapper, realm)
```

Comment:
I think the storing of the response from create_component into mapper is confusing, because the variable, mapper is never used, (nor does it update the mapper within the array, `updated_mappers`.

Reply:
Is there an intent that the contents of `update_mappers` be updated with the new values after these statements? If so, then there should something like, `mappers.update(...)` within this block of code.

From my simple test simply updating the content of the variable containing the value in the iteration does not update the item within the array.  For example:
```
x = [{"a": 1}, {"a": 2}, {"a": 3}, {"a": 4}]
for m in x:
    m = {"c": m["a"] + 4}
print(x)
```
results in:
```
[{'a': 1}, {'a': 2}, {'a': 3}, {'a': 4}]
```
(i.e. the contents of the array is not updated). However:
```
x = [{"a": 1}, {"a": 2}, {"a": 3}, {"a": 4}]
for m in x:
    m.update({"c": m["a"] + 4, "a": m["a"] - 1})
print(x)
```
results in:
```
[{'a': 0, 'c': 5}, {'a': 1, 'c': 6}, {'a': 2, 'c': 7}, {'a': 3, 'c': 8}]
```
(i.e. updated values)

So if the intent of updating the contents of `mapper` in line 940 was to update the value in `updated_mappers`, I do not think this actually will occur.  If the response from `update_component` includes a dict of the new object state, then it may be desirable to store that too.

Reply:
Thanks for your comments @pmdumuid. AFAIK, @laurpaum wrote this code in https://github.com/ansible-collections/community.general/commit/2589e9a0300ddd78730eb03f10c5d89e844463f5. You see it as a `+` in this diff because I change the first parameter name from `mapper` to `new_mapper` but I actually did not change the way the variable `mapper` is assigned (cf. https://github.com/ansible-collections/community.general/commit/bb8d3da7bd96576adf82d167fc9d5a51c8d4c5b0?diff=unified#diff-d215592986d2a3636f4120b8c8eeaaec937c7544bfafe9e1a1a7d78abea1611dL927-R940).

Can you please reconsider your review? If there is indeed a bug, it can be fixed in another pull request. There is no regression.

Reply:
Hi @jlamur -  , no problem, happy to have this consider resolved then.  My comment is fairly minor and mostly concerns readability.